src/odemis/dataio/tiff.py

```diff
@@ -2613,3 +2628,93 @@ def _iterDirectories(tiff_file):
             tiff_file.ReadDirectory()
             dir_index += 1
             yield dir_index
+
+
+### ThermoFisher Metadata parsing
+def _convertTFStoOdemisMD(metadata: str) -> dict:
+    """
+    Converts ThermoFisher Scientific (TFS) metadata to odemis metadata format
+    :param metadata: metadata string as ini (TFS format)
+    :return: (dict) metadata as dictionary (odemis format)
+    :raises: (KeyError): if the metadata cannot be parsed
+    """
+
+    # parse ini metadata as dictionary
+    fei_md = {}
+    try:
+        cfg = configparser.ConfigParser()
+        cfg.read_string(metadata)
+        fei_md = cfg
+    except Exception as e:
+        logging.debug("Failed to parse metadata as ini: %s", e)
```

Comment:
Probably worthy more than a debug message. I'd go for a  warning, or even exception.

Also, please immediately `return md`, so that this is not followed by every metadata read raising an exception.